rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,62 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/)
+([#community:nixos.org][1]) as
+our official community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space ([#community:nixos.org][1]) that has already been created which you
+can join by visting the [invite
+link][1].
+
+Update any other places that point to Freenode to point to Matrix
+instead.
+
+Port any bots that might be necessary such as:
+
+* nix-channel-monitor
+* github PR events
+* karma bot
+* eval bot
+* factoids bot
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+People are used to Freenode and it might be harder for people to use
+Matrix, which requires more complex clients.
+
```

Comment:
```suggestion

Plain text is easy to log to a file; to copy around; to pastebin; to store in any number of cloud services; and to read, and to grep, even in the absence of one's chat client. The richer format of Matrix messages may make these tasks fundamentally more difficult, or perhaps more reliant on cloud services.
```

Relatedly: For freenode we have had @samueldr's public logs. How will those be migrated to serve Matrix, or what will replace them? If they're replaced by something offered by Matrix (e.g., @kevincox suggests "We can allow guests to view"), do we know...
* What is the log retention policy? Is it anything less than @samueldr's? I hear Matrix.org is too overburdened to retain logs long-term.
* Does the replacement have the features of the existing logs (search, linking to a specific message)?

Reply:
There is https://github.com/matrix-org/matrix-static for logging which we can supposedly just run our servers or use the hosted instance of matrix: https://view.matrix.org/room/%21sgkZKRutwatDMkYBHU:nixos.org/

Reply:
xref https://github.com/NixOS/rfcs/pull/94#issuecomment-845321053

Reply:
You can also directly link to messages https://matrix.to/#/!sgkZKRutwatDMkYBHU:nixos.org/$iQg3aR-SfW5hw3XyjuZaqIT5LJummMMWfuRE0EQjTzQ?via=nixos.org&via=matrix.org&via=fairydust.space